Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -508,10 +510,84 @@ extension BrowserViewController: TopToolbarDelegate {
         self.present(container, animated: true)
       }
     }
+    
+    shields.showSubmitReportView = { [weak self] shieldsViewController in
+      shieldsViewController.dismiss(animated: true) {
+        guard let url = shieldsViewController.tab.url else { return }
+        self?.showSubmitReportView(for: url, isPrivateMode: shields.tab.isPrivate)
+      }
+    }
+    
     let container = PopoverNavigationController(rootViewController: shields)
     let popover = PopoverController(contentController: container, contentSizeBehavior: .preferredContentSize)
     popover.present(from: topToolbar.locationView.shieldsButton, on: self)
   }
+  
+  private func showSubmitReportView(for url: URL, isPrivateMode: Bool) {
+    // Strip fragments and query params from url
+    var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+    components?.fragment = nil
+    components?.queryItems = nil
+    guard let cleanedURL = components?.url else { return }
+    
+    let viewController = UIHostingController(rootView: SubmitReportView(
+      url: cleanedURL,
+      dismiss: { [weak self] in
+        self?.navigationController?.dismiss(animated: true)
+      },
```

Comment:
You should be able to just use the `dismiss` environment value inside your SwiftUI view hierarchy to dismiss

Reply:
Done